op-dispute-mon/mon/withdrawals_test.go

```diff
@@ -0,0 +1,83 @@
+package mon
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/contracts"
+	monTypes "github.com/ethereum-optimism/optimism/op-dispute-mon/mon/types"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	weth1 = common.Address{0x1a}
+	weth2 = common.Address{0x2b}
+)
+
+func makeGames() []*monTypes.EnrichedGameData {
+	weth1Balance := big.NewInt(4200)
+	weth2Balance := big.NewInt(6000)
+	withdrawalRequests := make(map[common.Address]*contracts.WithdrawalRequest)
+	withdrawalRequests[common.Address{0x01}] = &contracts.WithdrawalRequest{
+		Amount: big.NewInt(3),
+	}
+	withdrawalRequests[common.Address{0x02}] = &contracts.WithdrawalRequest{
+		Amount: big.NewInt(1),
+	}
+	game1 := &monTypes.EnrichedGameData{
+		Credits: map[common.Address]*big.Int{
+			common.Address{0x01}: big.NewInt(3),
+			common.Address{0x02}: big.NewInt(1),
+		},
+		WithdrawalRequests: withdrawalRequests,
+		WETHContract:       weth1,
+		ETHCollateral:      weth1Balance,
+	}
+	game2 := &monTypes.EnrichedGameData{
+		Credits: map[common.Address]*big.Int{
+			common.Address{0x01}: big.NewInt(46),
+			common.Address{0x02}: big.NewInt(1),
+		},
+		WithdrawalRequests: withdrawalRequests,
+		WETHContract:       weth2,
```

Comment:
Would be worth adding a third game here that shares a WETHContract with one of the others.